### ISSUE TITLE
can not find the all reference when c++ file extension is not `.cpp`

### ISSUE THREAD

**Issue by heartacker** [microsoft/vscode-cpptools#8687]
Bug type: Language Service

**Describe the bug**
版本: 1.64.0-insider (system setup)
提交: 23aa7ad51c95d90d48b1d352a316d7f0c2c4310d
日期: 2022-01-18T09:44:38.476Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.14393
C/C++: 1.8.0-INSIDER

**Steps to reproduce**

1. a `foo.h` file
2. a `foo.pln` file witch code is cpp
3. change the `.pln` file association with cpp
4. focus a function declaration in the header file, execute command `find all reference`
5. we could not see any reference in pln file
6. open the pln with vscode
7. execute command `find all reference` again
8. we could see that reference in pln file

**Expected behavior**
we could see reference in pln file

**Screenshots**

![image](https://user-images.githubusercontent.com/1876302/150071585-f9feab41-8262-4bae-8cc8-a3f1c971977b.png)
- before open pln file in vscode
![image](https://user-images.githubusercontent.com/1876302/150071689-7e90b018-b28f-40a7-bc3a-3bbd994877b4.png)

- after open pln file in vscode
![image](https://user-images.githubusercontent.com/1876302/150071651-ecc4f2d3-5ba7-41cc-9a14-a33fa24e88ec.png)

**Additional context**

`pln` is a special file extension of cpp in my solution

**Comment by heartacker** [comment]
```
"files.associations": {
    "*.pln": "cpp",
}
```

**Comment by sean-mcmanus** [comment]
The bug is that changing the files.association doesn't automatically trigger a Rescan. You can workaround it via running the C/C++: Rescan Workspace command.

**Comment by heartacker** [comment]
oh, thanks, BTW, I have associated plb with cpp before

**Comment by sean-mcmanus** [comment]
We believe this is fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/1.8.1 .

**Comment by heartacker** [comment]
seems that we still could not find the reference

**Comment by sean-mcmanus** [comment]
Sorry about that -- you are correct. I'll look into why the fix isn't working...

**Comment by sean-mcmanus** [comment]
I'm not reproing this with 1.8.4. I see "cpptools/didChangeSettings" after changing the files.associations and the .pln files get tag parsed and then found by Find All References. If you're still reproing this, do you have any idea what could be different with my repro and what logging do you see (with C_Cpp.loggingLevel set to "Debug")?

**Comment by heartacker** [comment]
![image](https://user-images.githubusercontent.com/1876302/153743049-cc795820-55ec-4e5f-98ad-854a2b8e4c5a.png)

```
  ...
  标记分析文件: C:\CRAFT\include\uti\VAR_BASE.H
  标记分析文件: C:\CRAFT\include\vi100lib\VI100_FIRMWARE.H
  标记分析文件: C:\CRAFT\include\vi100lib\vi100_reg.h
  正在分析: 已处理 590 个文件
“分析剩余文件”操作已完成。
cpptools/findAllReferences: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h
cpptools/finishUpdateSquiggles
错误波形曲线计数: 0
更新 IntelliSense 时间(秒): 49.496
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 15)
[警告] 某些引用可能丢失，因为在 查找所有引用 启动时，工作区分析不完整。

cpptools/getFoldingRanges: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 16)
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 17)
cpptools/textEditorSelectionChange
cpptools/activeDocumentChange: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 18)
cpptools/findAllReferences: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 19)
已取消请求: 19
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 20)
已取消请求: 20
```

![image](https://user-images.githubusercontent.com/1876302/153743056-faded02c-ce64-4ccc-be01-cd5deba2d348.png)
```
cpptools/textEditorSelectionChange
cpptools/textEditorSelectionChange
cpptools/activeDocumentChange: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 21)
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 22)
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 23)
cpptools/getSemanticTokens: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.pln (id: 24)
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.pln (id: 25)
textDocument/didOpen: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.pln
正在检查语法错误: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.pln
cpptools/textEditorSelectionChange
cpptools/getDocumentSymbols: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.pln (id: 26)
cpptools/textEditorSelectionChange
cpptools/activeDocumentChange: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.pln
cpptools/getDocumentSymbols
  标记分析文件: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.pln
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.pln (id: 27)
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.pln (id: 28)
IntelliSense 客户端不可用，请使用标记分析器转到定义。
IntelliSense 客户端不可用，请使用标记分析器获取快速信息。
使用标记分析器获取快速信息
  标记分析文件: F:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.30.30705\include\string
无法解析包含路径: PAT/pin_init.h
  标记分析文件: D:\ATC_Project\AC7840\chroma-ac7840-hqa\Logger.h
  标记分析文件: F:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.30.30705\include\numeric
  标记分析文件: F:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.30.30705\include\cstdarg
  标记分析文件: C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\ucrt\direct.h
  标记分析文件: F:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.30.30705\include\fstream
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.pln (id: 29)
cpptools/getFoldingRanges: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.pln (id: 30)
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.pln (id: 31)
IntelliSense 客户端不可用，请使用标记分析器转到定义。
IntelliSense 客户端不可用，请使用标记分析器获取快速信息。
使用标记分析器获取快速信息
正在为 D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.pln 发送编译参数
  包括: F:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\VC\TOOLS\MSVC\14.30.30705\INCLUDE
  包括: F:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\VC\TOOLS\MSVC\14.30.30705\ATLMFC\INCLUDE
  包括: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\UM
  包括: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\UCRT
  包括: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\SHARED
  包括: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\WINRT
  包括: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\CPPWINRT
  包括: C:\CRAFT\INCLUDE
  定义: _DEBUG
  定义: UNICODE
  定义: _UNICODE
  定义: __VSCODE__
  定义: CRAFT_3380
  预包含: cpp.hint
  预包含: C:/CRAFT/include/statement/TMU_CLASS.H
  预包含: C:/CRAFT/include/stnlib/defaultvalue.h
  预包含: C:/CRAFT/include/stnlib/stnall.h
  预包含: C:/CRAFT/include/stnlib/stnawilib.h
  预包含: C:/CRAFT/include/stnlib/stndpslib.h
  预包含: C:/CRAFT/include/stnlib/stnfluvi.h
  预包含: C:/CRAFT/include/stnlib/stnlevellib.h
  预包含: C:/CRAFT/include/stnlib/stnlib.h
  预包含: C:/CRAFT/include/stnlib/stnosclib.h
  预包含: C:/CRAFT/include/stnlib/stnotherlib.h
  预包含: C:/CRAFT/include/stnlib/stnpatlib.h
  预包含: C:/CRAFT/include/stnlib/stnpmulib.h
  预包含: C:/CRAFT/include/stnlib/stnrelaylib.h
  预包含: C:/CRAFT/include/stnlib/stnrf.h
  预包含: C:/CRAFT/include/stnlib/stnshmoo.h
  预包含: C:/CRAFT/include/stnlib/stnshmoo_old.h
  预包含: C:/CRAFT/include/stnlib/stnstatementid.h
  预包含: C:/CRAFT/include/stnlib/stntmu.h
  预包含: C:/CRAFT/include/uti/fwlib.h
  stdver: ms_c++14
  intelliSenseMode: windows-msvc-x64
正在为以下项的翻译单元中文件的 IntelliSense 更新进行排队: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.pln
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 32)
cpptools/textEditorSelectionChange
cpptools/textEditorSelectionChange
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 33)
cpptools/activeDocumentChange: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 34)
textDocument/documentHighlight: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 35)
cpptools/textEditorSelectionChange
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 36)
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 37)
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 38)
空闲环路: 重新分析活动文档
正在检查语法错误: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h
正在为以下项的翻译单元中文件的 IntelliSense 更新进行排队: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h
cpptools/finishUpdateSquiggles
错误波形曲线计数: 0
cpptools/findAllReferences: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h
cpptools/getSemanticTokens: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.pln (id: 39)
cpptools/getFoldingRanges: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 40)
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 41)
已取消请求: 41
cpptools/finishUpdateSquiggles
错误波形曲线计数: 88
更新 IntelliSense 时间(秒): 13.672
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 42)
已取消请求: 42
cpptools/getFoldingRanges: D:\ATC_Project\AC7840\chroma-ac7840-hqa\MCU.h (id: 43)

```

**Comment by sean-mcmanus** [comment]
It seems like you're hitting a different issue with `D:\ATC_Project\AC7840\chroma-ac7840-hqa` not in your browse path. What logging do you see when you run "C/C++: Rescan Workspace"? You need to see "Processing folder (non-recursive): D:\ATC_Project\AC7840\chroma-ac7840-hqa" or the files in that folder won't be tag parsed until they're opened. What is your includePath or browse.path setting or do you have configurationProvider or compileCommands set?

**Comment by heartacker** [comment]
<Summary>

```
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 46)
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 47)
cpptools/textEditorSelectionChange
textDocument/documentHighlight: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 48)
cpptools/textEditorSelectionChange
cpptools/activeDocumentChange: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h
cpptools/textEditorSelectionChange
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 49)
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 50)
textDocument/documentHighlight: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 51)
cpptools/textEditorSelectionChange
cpptools/textEditorSelectionChange
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 52)
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 53)
cpptools/findAllReferences: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln (id: 54)
textDocument/didOpen: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln
正在检查语法错误: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln
cpptools/textEditorSelectionChange
cpptools/getDocumentSymbols: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln (id: 55)
cpptools/getSemanticTokens: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln (id: 56)
cpptools/getDocumentSymbols
cpptools/activeDocumentChange: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln
  标记分析文件: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln
cpptools/getDocumentSymbols: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln (id: 57)
  标记分析文件: C:\CRAFT\include\firmware\3380_Board_Name.h
  标记分析文件: C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\shared\sal.h
已取消请求: 55
cpptools/getDocumentSymbols
无法解析包含路径: PAT/pin_init.h
cpptools/getFoldingRanges: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln (id: 58)
  标记分析文件: C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\shared\concurrencysal.h
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 59)
cpptools/textEditorSelectionChange
cpptools/textEditorSelectionChange
textDocument/documentHighlight: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 60)
cpptools/activeDocumentChange: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 61)
cpptools/getFoldingRanges: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 62)
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 63)
已取消请求: 63
cpptools/textEditorSelectionChange
cpptools/textEditorSelectionChange
空闲环路: 重新分析活动文档
正在检查语法错误: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h
正在为以下项的翻译单元中文件的 IntelliSense 更新进行排队: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 64)
cpptools/finishUpdateSquiggles
错误波形曲线计数: 0
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 65)
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 66)
正在为 D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln 发送编译参数
  包括: F:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\VC\TOOLS\MSVC\14.30.30705\INCLUDE
  包括: F:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\VC\TOOLS\MSVC\14.30.30705\ATLMFC\INCLUDE
  包括: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\UM
  包括: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\UCRT
  包括: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\SHARED
  包括: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\WINRT
  包括: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\CPPWINRT
  包括: C:\CRAFT\INCLUDE
  定义: _DEBUG
  定义: UNICODE
  定义: _UNICODE
  定义: __VSCODE__
  定义: CRAFT_3380
  预包含: cpp.hint
  预包含: C:/CRAFT/include/statement/TMU_CLASS.H
  预包含: C:/CRAFT/include/stnlib/defaultvalue.h
  预包含: C:/CRAFT/include/stnlib/stnall.h
  预包含: C:/CRAFT/include/stnlib/stnawilib.h
  预包含: C:/CRAFT/include/stnlib/stndpslib.h
  预包含: C:/CRAFT/include/stnlib/stnfluvi.h
  预包含: C:/CRAFT/include/stnlib/stnlevellib.h
  预包含: C:/CRAFT/include/stnlib/stnlib.h
  预包含: C:/CRAFT/include/stnlib/stnosclib.h
  预包含: C:/CRAFT/include/stnlib/stnotherlib.h
  预包含: C:/CRAFT/include/stnlib/stnpatlib.h
  预包含: C:/CRAFT/include/stnlib/stnpmulib.h
  预包含: C:/CRAFT/include/stnlib/stnrelaylib.h
  预包含: C:/CRAFT/include/stnlib/stnrf.h
  预包含: C:/CRAFT/include/stnlib/stnshmoo.h
  预包含: C:/CRAFT/include/stnlib/stnshmoo_old.h
  预包含: C:/CRAFT/include/stnlib/stnstatementid.h
  预包含: C:/CRAFT/include/stnlib/stntmu.h
  预包含: C:/CRAFT/include/uti/fwlib.h
  stdver: ms_c++14
  intelliSenseMode: windows-msvc-x64
正在为以下项的翻译单元中文件的 IntelliSense 更新进行排队: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 67)
cpptools/getFoldingRanges: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 68)
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 69)
已取消请求: 69
cpptools/findAllReferences: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 70)
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 71)
cpptools/finishUpdateSquiggles
错误波形曲线计数: 209
更新 IntelliSense 时间(秒): 7.311
cpptools/getFoldingRanges: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 72)
cpptools/textEditorSelectionChange
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 73)
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln (id: 74)
cpptools/textEditorSelectionChange
cpptools/textEditorSelectionChange
cpptools/activeDocumentChange: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln
cpptools/getFoldingRanges: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln (id: 75)
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln (id: 76)
空闲环路: 重新分析活动文档
正在检查语法错误: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln
正在为以下项的翻译单元中文件的 IntelliSense 更新进行排队: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln
cpptools/finishUpdateSquiggles
错误波形曲线计数: 209
更新 IntelliSense 时间(秒): 0.293
cpptools/getFoldingRanges: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln (id: 77)
cpptools/textEditorSelectionChange
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln (id: 78)
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 79)
cpptools/textEditorSelectionChange
cpptools/textEditorSelectionChange
cpptools/activeDocumentChange: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 80)
cpptools/getFoldingRanges: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 81)
空闲环路: 重新分析活动文档
正在检查语法错误: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h
正在为以下项的翻译单元中文件的 IntelliSense 更新进行排队: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h
正在为以下项的翻译单元中文件的 IntelliSense 更新进行排队: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.pln
cpptools/finishUpdateSquiggles
错误波形曲线计数: 0
cpptools/finishUpdateSquiggles
错误波形曲线计数: 209
更新 IntelliSense 时间(秒): 0.299
cpptools/getFoldingRanges: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 82)
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 83)
textDocument/documentHighlight: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 84)
cpptools/textEditorSelectionChange
cpptools/textEditorSelectionChange
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 85)
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 86)
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 87)
cpptools/findAllReferences: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h
textDocument/documentHighlight: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 88)
cpptools/textEditorSelectionChange
cpptools/textEditorSelectionChange
cpptools/getCodeActions: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 89)
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 90)
cpptools/findAllReferences: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h
textDocument/hover: D:\ATC_Project\AC7840\chroma-ac7840-hqa\LevelsTimming.h (id: 91)

```
</Summary>


```json
{
    "configurations": [
        {
            "name": "Chroma_cpp",
            "includePath": [
                "${default}",
                "${workspaceFolder}/**",
                "C:\\CRAFT\\include\\**"
            ],
            "defines": [
                "_DEBUG",
                "UNICODE",
                "_UNICODE",
                "__VSCODE__",
                "CRAFT_3380"
            ],
            "intelliSenseMode": "windows-msvc-x64",
            // "compilerPath": "C:\\Program Files (x86)\\Microsoft Visual Studio 12.0\\VC\\bin\\x86_amd64\\cl.exe",
            "browse": {
                "path": [
                    "C:\\CRAFT\\include\\"
                ]
            },
            "forcedInclude": [
                "${default}",
                "cpp.hint",
                //管脚的定义
                "PAT\\pin_init.h",
                //
                "C:\\CRAFT\\include\\statement\\TMU_CLASS.H", //
                // stnlib
                "C:\\CRAFT\\include\\stnlib\\defaultvalue.h",
                "C:\\CRAFT\\include\\stnlib\\stnall.h",
                "C:\\CRAFT\\include\\stnlib\\stnawilib.h",
                "C:\\CRAFT\\include\\stnlib\\stndpslib.h",
                "C:\\CRAFT\\include\\stnlib\\stnfluvi.h",
                "C:\\CRAFT\\include\\stnlib\\stnlevellib.h",
                "C:\\CRAFT\\include\\stnlib\\stnlib.h",
                "C:\\CRAFT\\include\\stnlib\\stnosclib.h",
                "C:\\CRAFT\\include\\stnlib\\stnotherlib.h",
                "C:\\CRAFT\\include\\stnlib\\stnpatlib.h",
                "C:\\CRAFT\\include\\stnlib\\stnpmulib.h",
                "C:\\CRAFT\\include\\stnlib\\stnrelaylib.h",
                "C:\\CRAFT\\include\\stnlib\\stnrf.h",
                "C:\\CRAFT\\include\\stnlib\\stnshmoo.h",
                "C:\\CRAFT\\include\\stnlib\\stnshmoo_old.h",
                "C:\\CRAFT\\include\\stnlib\\stnstatementid.h",
                "C:\\CRAFT\\include\\stnlib\\stntmu.h",
                // 枚举的定义
                "C:\\CRAFT\\include\\uti\\fwlib.h"
            ],
            "cppStandard": "c++14"
            // "configurationProvider": "ms-vscode.makefile-tools"
        }
    ],
    "version": 4
}
```

**Comment by heartacker** [comment]
yes, I can repro this problem after rename my code to `cpp` in stead of `pln`.

so maybe there is another problem

**Comment by sean-mcmanus** [comment]
Yeah, I believe you're hitting https://github.com/microsoft/vscode-cpptools/issues/8851, which we either need procmon info or tag parser error info (pending 1.9.2) to diagnose further.

We'll keep this issue to track the bug that was fixed, i.e. "changing the files.association doesn't automatically trigger a Rescan".

**Comment by Colengms** [comment]
It looks like this was fixed in 1.8?  Closing as fixed.

**Comment by sean-mcmanus** [comment]
Actually, it should be fixed with 1.9.6 though.
